### PR TITLE
Nodejs-week2

### PR DESCRIPTION
**Pull request by YuejiaoShi**
**[Warmup Requirements](https://github.com/HackYourFuture-CPH/node.js/tree/main/week2/homework#warmup)**

**Build a search engine with 3 routes:**

- GET /search
- GET /documents/:id
- POST /search

### CODE REVIEW

File: nodejs/week2/app.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    const documents = JSON.parse(contents);
+
+    if (!q) {
+      return res.json(documents);
+    } else {
+      const filteredDocs = documents.filter((doc) =>
+        Object.values(doc).some(
+          (value) =>
+            value !== undefined &&
+            value.toString().toLowerCase().includes(q.toLowerCase())
```
Comment (magdazelena) [inline nodejs/week2/app.js]: very nice, also with making this case-agnostic 🤩

File: nodejs/week2/app.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+// POST /search
+app.post("/search", async (req, res) => {
+  const { q } = req.query;
+  const { fields } = req.body;
+
+  if (q && fields) {
+    return res
+      .status(400)
+      .send("Cannot use both query parameter and request body.");
+  }
```
Comment (magdazelena) [inline nodejs/week2/app.js:86]: I see, so there are two options available. In that case I would keep the conditions logic in particular endpoints and extract the actual functions (below) to separate utility functions. You know, to keep stuff DRY (Don't Repeat Yourself) :)

**Review by magdazelena — APPROVED** [review]
great work! I left a comment, but it's more a note than requirement for change :)

File: nodejs/week2/app.js (hunk 2 above)
Comment (YuejiaoShi) [reply nodejs/week2/app.js:86]: I've already extract the functions into separate utilities. You're absolutely right, it's good clean code practice :D. Thank u so much for your review!🫶

**Review by magdazelena — APPROVED** [review]
fantastic work :) I love your granular commits, it makes it sooooo easy to review and it's a great practice as well 👏

**Pull request merged**